src/components/burger-menu/burgerMenu.jsx

```diff
@@ -1,15 +1,45 @@
-import "./burgerMenu.css";
+import { useRef } from "react";
+import style from "../nav/Navbar.module.css";
 
-const BurgerMenu = () => {
+const BurgerMenu = ({ navRef }) => {
+  const menuIconRef = useRef(null);
+  const closeIconRef = useRef(null);
+
+  const MenuClick = () => {
+    if (navRef.current && menuIconRef.current && closeIconRef.current) {
+      navRef.current.classList.add(style.navVisible);
+      menuIconRef.current.classList.add(style.menuHidden);
+      closeIconRef.current.classList.add(style.closeVisible);
+
+      menuIconRef.current.classList.remove(style.menuVisible);
+      closeIconRef.current.classList.remove(style.closeHidden);
+      navRef.current.classList.remove(style.navHidden);
+      document.body.style.overflow = "hidden";
+    }
+  };
+  const CloseClick = () => {
+    if (navRef.current && menuIconRef.current && closeIconRef.current) {
+      navRef.current.classList.add(style.navHidden);
+      menuIconRef.current.classList.add(style.menuVisible);
+      closeIconRef.current.classList.add(style.closeHidden);
+
+      navRef.current.classList.remove(style.navVisible);
+      menuIconRef.current.classList.remove(style.menuHidden);
+      closeIconRef.current.classList.remove(style.closeVisible);
+      document.body.style.overflow = "visible";
+    }
+  };

```

Comment:
ai mare grija cand folosesti `useRef` pentru ca react lucreaza cu virtual DOM, si tu cand folosesti ref targetezi the real DOM, e f periculos ce ai facut aici. 

in cazul acesta trebuia sa folosesti `useState` .. si sa plimbi callbacku pana intr-un loc comun cum ti am aratat data trecuta inc all